src/entitysdk/util.py

```diff
@@ -83,6 +83,11 @@ def stream_paginated_request(
     Returns:
         An iterator of dicts.
     """
+    if has_limit := (limit is not None):
+        limit = int(limit)
+        if limit <= 0:
+            raise EntitySDKError("Limit must be either None or strictly positive.")
+
```

Comment:
Wouldn't be enough this?
```suggestion
    if limit is not None and limit <= 0:
        raise EntitySDKError("Limit must be either None or strictly positive.")

```

Reply:
I am casting the limit mainly to constrain what the user may pass here. Floats, like 0.5, can mess up the logic in this function.